executor/jwt_authenticator.go

```diff
@@ -179,3 +196,76 @@ type OpenIDConfiguration struct {
 	Issuer  string `json:"issuer"`
 	JWKSURI string `json:"jwks_uri"`
 }
+
+func getFnName() (string, error) {
+	name, ok := os.LookupEnv("OPENFAAS_NAME")
+	if !ok || len(name) == 0 {
+		return "", fmt.Errorf("env variable 'OPENFAAS_NAME' not set")
+	}
+
+	return name, nil
+}
+
+func getFnNamespace() (string, error) {
```

Comment:
How would we run this with go run ?

Reply:
I made the path used to lookup the namespace configurable by setting the env variable `namespace_mount_path`. This allows you to set any namespace name when running a function locally with Docker or `faas-cli local-run`

e.g for local testing the watchdog with jwt_auth:

```sh
fprocess="env" \
mode="serializing" \
jwt_auth=true \
jwt_auth_local=true \
namespace_mount_path=./secrets/namespace \
OPENFAAS_NAME=env \
./bin/fwatchdog-amd64
```

Reply:
I think I'd prefer it to be `OPENFAAS_NAMESPACE` as a priority order.

Reply:
This has been changed. The env variable `OPENFAAS_NAMESPACE` is now used to lookup the namespace if it is not set the watchdog will try to read the namespace from `/var/run/secrets/kubernetes.io/serviceaccount/namespace`.